testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/CliCompletionTestCase.java

```diff
@@ -479,29 +482,41 @@ public void testAppendCustomFileRelativeDirCompletion() throws Exception {
     }
 
     @Test
-    @Ignore("Un-ignore when WFCORE-3561 is fixed")
     public void testAppendCustomFileAbsoluteDirCompletion() throws Exception {
         Path tempFile = Files.createTempFile("tempFile", ".tmp");
         String tempFileStringPath = tempFile.toString();
-
         try {
             ctx.handle("version >" + tempFileStringPath);
             List<String> paths = new ArrayList<>();
-            for (Path path: tempFile.toAbsolutePath()) {
-                paths.add(path.toFile().getName());
+
+            // Split the absolute path. First item must be the root directory (for X:\ on Windows)
+            paths.add(tempFile.toAbsolutePath().getRoot().toString());
+            for (int i=0; i < tempFile.toAbsolutePath().getNameCount(); i++) {
+                paths.add(tempFile.toAbsolutePath().getName(i).toString());
             }
 
             String cmd = "version >>";
             String testPath = "";
             List<String> candidates = new ArrayList<>();
-            for (int i = 0; i < paths.size(); i++) {
+            // Iterate through whole path except the last element which is the filename itself
+            for (int i = 0; i < (paths.size() - 1); i++) {
                 String p = paths.get(i);
-                testPath += File.separator + p;
-                candidates = complete(ctx, cmd + testPath, null);
-                if (i + 1 < paths.size()) {
-                    assertTrue(candidates.toString(), candidates.contains(p + File.separator));
+                if (i == 0 || (i == paths.size() - 1)) {
+                    testPath += p;
+                } else {
+                    testPath += p + File.separator;
+                }
+
+                // Candidate should be directory
+                if (i < tempFile.toAbsolutePath().getNameCount() - 1) {
+                    candidates = complete(ctx, cmd + testPath, null);
```

Comment:
You could extract this line and put it prior to the branches. It is common on the 2 branches.